qt/scientific_interfaces/Indirect/IndirectFitPropertyBrowser.cpp

```diff
@@ -0,0 +1,394 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "IndirectFitPropertyBrowser.h"
+#include "FunctionTemplateBrowser.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/MultiDomainFunction.h"
+#include "MantidAPI/WorkspaceGroup.h"
+
+#include "MantidQtWidgets/Common/FitOptionsBrowser.h"
+#include "MantidQtWidgets/Common/FunctionBrowser.h"
+#include "MantidQtWidgets/Common/SignalBlocker.h"
+
+#include <QAction>
+#include <QCheckBox>
+#include <QFormLayout>
+#include <QLabel>
+#include <QMenu>
+#include <QMessageBox>
+#include <QPushButton>
+#include <QSettings>
+#include <QSignalMapper>
+#include <QSplitter>
+#include <QStackedWidget>
+#include <QVBoxLayout>
+
+#include <iostream>
+
+using namespace Mantid::API;
+using namespace MantidQt::MantidWidgets;
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+/**
+ * Constructor
+ * @param parent :: The parent widget - must be an ApplicationWindow
+ * @param mantidui :: The UI form for MantidPlot
+ */
+IndirectFitPropertyBrowser::IndirectFitPropertyBrowser(QWidget *parent)
+    : QDockWidget(parent), m_templateBrowser(nullptr),
+      m_functionWidget(nullptr) {
+  setFeatures(QDockWidget::DockWidgetFloatable |
+              QDockWidget::DockWidgetMovable);
+  setWindowTitle("Fit Function");
+}
+
+void IndirectFitPropertyBrowser::initFunctionBrowser() {
+  m_functionBrowser = new FunctionBrowser(nullptr, true);
+  m_functionBrowser->setObjectName("functionBrowser");
+  // Process internally
+  connect(m_functionBrowser, SIGNAL(globalsChanged()), this,
+          SLOT(updateFitType()));
+  // Re-emit
+  connect(m_functionBrowser, SIGNAL(functionStructureChanged()), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser,
+          SIGNAL(parameterChanged(const QString &, const QString &)), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser, SIGNAL(tiesChanged()), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser, SIGNAL(constraintsChanged()), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser, SIGNAL(globalsChanged()), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser,
+          SIGNAL(localParameterButtonClicked(const QString &)), this,
+          SIGNAL(localParameterEditRequested(const QString &)));
+}
+
+void IndirectFitPropertyBrowser::initFitOptionsBrowser() {
+  m_fitOptionsBrowser = new FitOptionsBrowser(
```

Comment:
unique_ptr

Reply:
This is a child of m_functionWidget which is a child of the main class so the pointer lifetimes will be handled by the Qt parent child system